build_runner/lib/src/generate/build_impl.dart

```diff
@@ -103,7 +104,17 @@ class BuildImpl {
   Future<Null> _updateAssetGraph(Map<AssetId, ChangeType> updates) async {
     var deletes = _assetGraph.updateAndInvalidate(
         _buildActions, updates, _packageGraph.root.name);
-    await Future.wait(deletes.map(_delete));
+    await Future.wait(deletes.map((node) {
```

Comment:
I actually struggled a fair bit to come up with anything clean here - its harder than it seems and this was the best I could come up with.

If you have a better idea that doesn't involve a huge refactor let me know :).

Reply:
Actually there is one other option that I just thought of, we could change `updateAndInvalidate` to take in a `Future<Null> delete(AssetId id)` method, which it would call _before_ actually removing the node from the graph (and we would pass `_delete` here. That also means `updateAndInvalidate` would become async but I think that is fine, and it removes any of this hackery.

Reply:
Updated it to use the new logic, definitely cleaner IMO (at least we don't have to duplicate the cache location logic).